camayoc/qcs_models.py

```diff
@@ -42,43 +65,140 @@ def __ne__(self, other):
         """Return true if both objects are not equal."""
         return not self == other
 
+    def create(self, **kwargs):
+        """Send POST request to the self.endpoint of this object.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            ``request.request()`` method.
+
+        Sends ``self.payload()`` as the data of the POST request, thereby
+        creating an object on the server.
+
+        Before returning the requests.models.Response to the caller, the
+        ``_id`` of this object is set using the data from the response.
+
+        Returns a requests.models.Response. The json of this response contains
+        the data associated with this object's ``self._id``.
+        """
+        response = self.client.post(self.endpoint, self.payload(), **kwargs)
+        self._id = response.json()['id']
+        return response
+
+    def read_all(self, **kwargs):
+        """Send GET request to read all objects of this type.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+
+        Returns a requests.models.Response. The json of this response
+        contains a list of dictionaries with the data associated with each
+        object of this type stored on the server.
+        """
+        return self.client.get(self.endpoint, **kwargs)
+
+    def read_self(self, **kwargs):
+        """Send GET request to the self.endpoint/{id} of this object.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+
+        Returns a requests.models.Response. The json of this response contains
+        the data associated with this object's `self._id`.
+        """
+        return self.client.get(self.path(), **kwargs)
+
+    def update(self, **kwargs):
+        """Send PUT request to the self.endpoint/{id} of this object.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+
+        Sends `self.payload()` as the data of the PUT request, thereby updating
+        the object on the server with the same `id` as this object with the
+        fields contained in `self.payload()`.
+
+        Returns a requests.models.Response. The json of this response contains
+        the data associated with this object's `self._id`.
+        """
+        return self.client.put(self.path(), self.payload(), **kwargs)
+
+    def delete(self, **kwargs):
+        """Send DELETE request to the self.endpoint/{id} of this object.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+
+        Returns a requests.models.Response. A successful delete has the return
+        code `204`.
+        """
+        return self.client.delete(self.path(), **kwargs)
+
+    def assert_matches_server(self):
```

Comment:
I see this being useful but I don't think inside this class is the place for it. It seems like an util function